hw/ip/spi_host/doc/_index.md

```diff
@@ -660,16 +659,613 @@ To use this reset, assert {{< regref "CONTROL.SW_RST" >}}, and then wait for the
+
+The state machine itself easiest understood by first considering a simple case, with CSAAT set to zero.
+For this initial discussion it is assumed that every command consists of one single segment.
+Multi-segment commands are considered in the following sections.
+In this case the state machine can be simplified to the following figure.
+
+![](spi_host_fsm_simplified.svg)
+
+The operation of the state machine is the same regardless of the polarity (CPOL) or phase (CPHA) of the current command.
+Commands with `CPOL==0` or `CPOL==1` are processed nearly identically, since the only difference is in the polarity of the `sck` output.
+The state machine drives an internal `sck` clock signal, which is low except when the FSM is in the `InternalClockHigh` state.
+If `CPOL==0` this clock is registered as is to the external `sck` ports.
+If `CPOL==1` the internal clock is *inverted* before the final `sck` output register.
+
+In the following description of the individual states, it is assumed that `CPOL==0`.
+To understand the IP's behavior for transactions with `CPOL==1`, simply invert the value of `sck`.
+
+1. Idle state: In this initial reset state, The `sck` signal is low, and all `csb` lines are high (i.e., inactive).
+An input command is registered whenever `command_valid_i` and `command_ready_o` are both high (i.e., when the signal `new_command = command_valid_i & command_ready_o` is high), in which case the state machine transitions to the `WaitLead` state.
+
+2. WaitLead state: In this state, `sck` remains low, and the `csb` line corresponding to `csid` is asserted-low.
+The purpose of this state is to hold `sck` low for at least `csnlead` + 1 timeslices, before the first rising edge of `sck`.
+For his reason, the FSM uses the `wait_cntr` to track the number of timeslices spent in this state, and only exits when `wait_cntr` counts down to zero, at which point the FSM transitions to the `InternalClkHigh` state.
+
+3. InternalClkHigh state: Entering this state drives `sck` high.
+This state repeats many times per segment, and usually transitions to the `InternalClkLow` state.
+The FSM transitions to the `WaitTrail` state only when the entire segment has been transmitted/received (as indicated by the signals last_bit and last_byte).
+This state machine usually only lasts stays in this state for one timeslice, except when the FSM is disabled or stalled.
+
+4. InternalClkLow state: This state serves to drive `sck` low between visits to the `InternalClkHigh` state.
+This state always returns back to the `InternalClkHigh` state in the next timeslice.
+
+5. WaitTrail state: Similar to the WaitLead, this state serves to control the timing of the `csb` line.
+The FSM uses the `wait_cntr` register to ensure that it remains in this state for `csntrail+1` timeslices, during which time the active `csb` is still held low.
+The `wait_cntr` register resets to {{< regref "CONFIGOPT.CSNTRAIL" >}} upon entering this state, and is decremented once per timeslice.
```

Comment:
Should be CONFIGOPTS, I think. Similarly for the bullet below.

Reply:
```suggestion
The `wait_cntr` register resets to {{< regref "CONFIGOPTS.CSNTRAIL" >}} upon entering this state, and is decremented once per timeslice.
```